jerry-core/ecma/operations/ecma-objects.c

```diff
@@ -2474,6 +2474,66 @@ ecma_object_class_is (ecma_object_t *object_p, /**< object */
   return false;
 } /* ecma_object_class_is */
 
+/**
+ * Checks if the given argument is an array or not.
+ *
+ * @return bool
+ */
+inline bool JERRY_ATTR_ALWAYS_INLINE
+ecma_object_is_array (ecma_value_t arg) /**< argument */
+{
+  if (ecma_is_value_object (arg))
+  {
+    if (ecma_get_object_type (ecma_get_object_from_value (arg)) == ECMA_OBJECT_TYPE_ARRAY)
+    {
+      return true;
+    }
+  }
+
+  return false;
+} /* ecma_object_is_array */
+
+#if ENABLED (JERRY_ES2015_BUILTIN)
+/**
+ * Object's IsConcatSpreadable operation
+ *
+ * See also:
+ *          ECMA-262 v6, 22.1.3.1.1;
+ *
+ * @return ecma_value_t
+ */
+ecma_value_t
+ecma_object_is_concat_spreadable (ecma_value_t arg) /**< argument */
+{
+  if (!ecma_is_value_object (arg))
+  {
+    return ECMA_VALUE_FALSE;
+  }
+
+  ecma_value_t spreadable = ecma_op_object_get_by_symbol_id (ecma_get_object_from_value (arg),
+                                                             LIT_MAGIC_STRING_IS_CONCAT_SPREADABLE);
+
+  if (ECMA_IS_VALUE_ERROR (spreadable))
+  {
+    return spreadable;
+  }
+
+  if (!ecma_is_value_undefined (spreadable))
+  {
+    return ecma_make_boolean_value (ecma_op_to_boolean (spreadable));
+  }
+
+  ecma_free_value (spreadable);
```

Comment:
At this point `spreadable` is `ECMA_VALUE_UNDEFINED` so no need to free the value.